### ISSUE TITLE
range pattern `1..` in a match incorrectly reported as an error 

### ISSUE THREAD

**Issue by keldonin** [intellij-rust/intellij-rust#8360]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4321-213
* **Rust toolchain version:** 1.56.1 x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2021.3 Community Edition (IC-213.5744.223)
* **Operating system:** macOS 11.6
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
It seems like the linter does not detect properly expressions like the following:
```rust
match whatever {
        1.. => { ... },
```

as it redlines the expression, and reports`<pat const> expected, got '=>'` when hovering on the issue with the mouse.

## Steps to reproduce

Put the following code into the editor to reproduce:

```rust
use std::env::args;

fn main() {
    let args = args();

    match args.len() {
        1.. => println!("BOOM"),
        _ => println!("PASS")
    };
}
```

the 7th line should feature red lining (see attached screen capture below:)

![2022-01-05_09-03-12](https://user-images.githubusercontent.com/1104707/148182790-2b8fc6a3-08e1-4a3e-b0cd-02a866091514.png)

**Comment by ortem** [comment]
@keldonin Thank you for the detailed report. 

Probably we haven’t implemented “RangeFrom” parsing rules yet (stabilized in Rust 1.55, https://github.com/rust-lang/rust/pull/83918)

**Comment by neonaot** [comment]
Fixed by https://github.com/intellij-rust/intellij-rust/pull/9808
